workers/data_refinery_workers/processors/test_salmon.py

```diff
@@ -98,11 +98,44 @@ def test_salmon(self):
         except FileNotFoundError:
             pass
 
-        job = prepare_job()
+        job, files = prepare_job()
         salmon.salmon(job.pk)
         job = ProcessorJob.objects.get(id=job.pk)
         self.assertTrue(job.success)
 
+    def test_fastqc(self):
+
+        job, og_files = prepare_job()
+        win_context = {
+            'job': job,
+            'job_id': 789,
+            'qc_directory': "/home/user/data_store/raw/TEST/SALMON/qc",
+            'original_files': og_files,
+            'success': True
+        }
+
+        # Ensure clean testdir
+        shutil.rmtree(win_context['qc_directory'], ignore_errors=True)
+        os.makedirs(win_context['qc_directory'], exist_ok=True)
+        win_context = salmon._prepare_files(win_context)
+
+        win = salmon._run_fastqc(win_context)
+        self.assertTrue(win['success'])
+        win = salmon._run_multiqc(win_context)
+        self.assertTrue(win['success'])
+
+        for file in win['qc_files']:
+            self.assertTrue(os.path.isfile(file.absolute_file_path))
+
+        fail_context = {
+            'job': job,
+            'job_id': 'hippityhoppity',
+            'qc_directory': "/home/user/data_store/raw/TEST/SALMON/derp",
+            'original_files': [],
```

Comment:
Is this empty list what causes this to fail?

Reply:
Yes, fastqc will fail without valid input, so this test checks the fail path appropriate sets the `success` keyvalue